src/main/java/com/karankumar/bookproject/BookProjectApplication.java

```diff
@@ -18,9 +18,10 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.autoconfigure.web.servlet.error.ErrorMvcAutoConfiguration;
+import org.springframework.boot.web.servlet.support.SpringBootServletInitializer; 
 
 @SpringBootApplication(exclude = ErrorMvcAutoConfiguration.class)
-public class BookProjectApplication {
+public class BookProjectApplication extends SpringBootServletInitializer {
```

Comment:
Why have you added this?

Reply:
it helps us to deploy in war or jar file instead of running main class as earlier. Here is some info about it https://www.baeldung.com/spring-boot-servlet-initializer